Add extension point to override the app logo and name

oltionchampari opened the pull request:
Closes Caleydo/ordino#319

### Summary

* Created extension point for the logo and the name
* Used  a `loadIcon` function because just passing the path string and then importing it in the code does not work
 this `import(icon)` does not work (as far as i tested) if icon is a variable


### Usage

```ts
  registry.push(EP_ORDINO_LOGO, 'ordino_logo', () => import('ordino/dist/assets/logos/ordino.svg').then(PluginRegistry.getInstance().asResource), <IOrdinoLogoDesc>{
    text: 'Ordino',
    width: 30,
    height: 30
  });
```

thinkh commented on src/phovea.ts:
```diff
@@ -99,6 +100,12 @@ export default function (registry: IRegistry) {
     priority: 30
   });
 
+  registry.push(EP_ORDINO_LOGO, 'ordino_logo', () => ({}), <IOrdinoLogoDesc>{
+    loadIcon() {
+      return import('ordino/dist/assets/logos/ordino.svg');
+    },
```
```suggestion
  registry.push(EP_ORDINO_LOGO, 'ordino_logo', function () {
    return import('ordino/dist/assets/logos/ordino.svg').then(PluginRegistry.getInstance().asResource), <IOrdinoLogoDesc>{
```

Could we use the loading mechanism similar to the locales (see the next extension point below) instead?

oltionchampari commented:
Changed the base branch to `bootstrap-v5`
From my tests,  the plugin registered from the app is at index `0` so i changed the code accordingly
Do you think this is always the case?

thinkh commented:
> From my tests, the plugin registered from the app is at index `0` so i changed the code accordingly
> Do you think this is always the case?

That depends on the import order in the workspace's phovea_registry.js file. It only works in this order:

```ts
import 'ordino_public/phovea_registry.js';
import 'ordino/phovea_registry.js';
// ...
```

thinkh reviewed (approved):
Thanks for the feature. It works in my tests.
